os/user/getent_unix.go

```diff
@@ -0,0 +1,111 @@
+// +build aix darwin dragonfly freebsd js,wasm !android,linux netbsd openbsd solaris
+// +build !cgo osusergo
+
+package user
+
+import (
+	"bytes"
+	"os"
+	"strconv"
+	"strings"
+)
+
+func iterateUsers(fn NextUserFunc) error{
+	f, err := os.Open(userFile)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+	_, err = readColonFile(f, usersIterator(fn))
+	return err
+}
+
+func iterateGroups(fn NextGroupFunc) error{
+	f, err := os.Open(groupFile)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+	_, err = readColonFile(f, groupsIterator(fn))
+	return err
+}
+
+// usersIterator does not return value other than error, as we want to make sure that we iterate through
+// entire file with readColonFile. If error is returned from fn - iteration process is terminated
+func usersIterator(fn NextUserFunc) lineFunc{
+	return func(line []byte) (interface{}, error) {
+		if u, ok := parseUser(line).(*User); ok{
+			err := fn(u)
+			if err != nil{
+				return nil, err
+			}
+		}
+		return nil, nil
+	}
+}
+
+// groupsIterator does not return value other than error, as we want to make sure that we iterate through
+// entire file with readColonFile. If error is returned from fn - iteration process is terminated
+func groupsIterator(fn NextGroupFunc) lineFunc{
+	return func(line []byte) (interface{}, error) {
+		if g, ok := parseGroup(line).(*Group); ok{
+			err := fn(g)
+			if err != nil{
+				return nil, err
+			}
+		}
+		return nil, nil
+	}
+}
+
+// Returns a *User from a line. If line can't be parsed correctly - nil will be returned.
+func parseUser (line []byte) interface{} {
```

Comment:
can `matchUserIndexValue` be refactored in a way to match the interface expected by `usersIterator`? There are quite a lot of similarities with that function, we should aim to consolidate them.